### ISSUE TITLE
Including kpatch-macros.h causes version disagreement

### ISSUE THREAD

**Issue by julien-thierry** [dynup/kpatch#1084]
The following patch is based on RHEL-7.7 codebase:

```
diff --git a/mm/gup.c b/mm/gup.c
index c89c0a4959f..85947880934 100644
--- a/mm/gup.c
+++ b/mm/gup.c
@@ -12,6 +12,8 @@

 #include "internal.h"

+#include <kpatch-macros.h>
+
 static int follow_pfn_pte(struct vm_area_struct *vma, unsigned long address,
                pte_t *pte, unsigned int flags)
 {
@@ -1038,6 +1040,8 @@ long populate_vma_page_range(struct vm_area_struct *vma,
        if ((vma->vm_flags & (VM_WRITE | VM_SHARED)) == VM_WRITE)
                gup_flags |= FOLL_WRITE;

+       barrier();
+
        /*
         * We want mlock to succeed for regions that have any permissions
         * other than PROT_NONE.```
```

kpatch-build successfully produces a livepatch module from it, but the module fails to load with the following error:
```
loading patch module: /var/lib/kpatch/3.10.0-1062.el7.x86_64/livepatch-0001-Test-ignore-function-patch.ko
insmod: ERROR: could not insert module /var/lib/kpatch/3.10.0-1062.el7.x86_64/livepatch-0001-Test-ignore-function-patch.ko: Invalid parameters
kpatch: failed to load module /var/lib/kpatch/3.10.0-1062.el7.x86_64/livepatch-0001-Test-ignore-function-patch.ko
```

Kernel logs contain the following:
```
[32763.834215] livepatch_0001_Test_ignore_function_patch: disagrees about version of symbol __get_user_pages
[32763.844944] livepatch_0001_Test_ignore_function_patch: Unknown symbol __get_user_pages (err -22)
```

The issue seems to occur when including kpatch-macros.h in a file where a modified function references an unmodified exported function defined in the same file. Causing the exported symbol to be in the symbol table of the livepatch module.

In the above patch, removing the include or modifying a function that doesn't directly reference an exported function will allow the livepatch module to be loaded successfully.

**Comment by sm00th** [comment]
> The issue seems to occur when including kpatch-macros.h in a file where a modified function references an unmodified exported function defined in the same file. Causing the exported symbol to be in the symbol table of the livepatch module.

It is actually included in both cases, it's checksum (calculated by modpost) changes in case when both hunks are there.

**Comment by sm00th** [comment]
Looked at it a bit more, the problem here is that __get_user_pages() uses VM_BUG_ON (and maybe others) which prints out __LINE__ it happened on. With the include on top of the file everything moves down, line number change and function's code changes slightly, which is enough to yield different checksum.

As a workaround for this specific patch you can move the include closer to where it is actually used.

**Comment by joe-lawrence** [comment]
Fixed by PR #1114 ("Fix symvers")
